clouddriver-sql/src/main/kotlin/com/netflix/spinnaker/clouddriver/sql/sql.kt

```diff
@@ -34,36 +30,29 @@ internal val taskStatesFields = listOf("id", "task_id", "created_at", "state", "
 internal val taskResultsFields = listOf("id", "task_id", "body").map { field(it) }
 
 /**
- * Run the provided [fn] in a transaction, retrying on failures using [retryProperties].
+ * Run the provided [fn] in a transaction, retrying on failures using resilience4j.retry.instances.sqlTransaction
+ * configuration.
  */
-@Deprecated("use transactional(retry, fn) instead")
-internal fun DSLContext.transactional(retryProperties: RetryProperties, fn: (DSLContext) -> Unit) {
-  retrySupport.retry({
-    transaction { ctx ->
-      fn(DSL.using(ctx))
-    }
-  }, retryProperties.maxRetries, retryProperties.backoffMs, false)
+@Retry(name = "sqlTransaction")
+internal fun DSLContext.transactional(fn: (DSLContext) -> Unit) {
+  transaction { ctx ->
+    fn(DSL.using(ctx))
+  }
 }
 
 /**
- * Run the provided [fn] in a transaction, retrying on failures using [retry].
+ * Run the provided [fn], retrying on failures using resilience4j.retry.instances.sqlRead configuration.
  */
-internal fun DSLContext.transactional(
-  retry: Retry,
-  fn: (DSLContext) -> Unit
-) {
-  retry.executeRunnable {
-    transaction { ctx ->
-      fn(DSL.using(ctx))
-    }
-  }
+@Retry(name = "sqlRead")
+internal fun <T> DSLContext.read(fn: (DSLContext) -> T): T {
+  return fn(this)
 }
 
 /**
- * Run the provided [fn] with retry support.
+ * Run the provided [fn], retrying on failures using resilience4j.retry.instances.sqlDelete configuration.
+ * TODO(csmalley):  Investigate if usage of this is necessary, or if delete calls can be transactional
```

Comment:
Yeah, we can use `transactional` for deletes. It's really "read" vs "write" connections (replicas vs master).